tests/test_git_changelog_by_date.py

```diff
@@ -0,0 +1,90 @@
+import os
+from datetime import datetime, timedelta
+
+from bs4 import BeautifulSoup
+from git import Repo
+
+from nose.tools import (
+    assert_equal,
+)
+
+from . import MakeTestableMixin, TempDirTestCase
+from sphinx_git import GitChangelogByDate
+
+
+class TestableGitChangelogByDate(MakeTestableMixin, GitChangelogByDate):
+
+    pass
+
+
+class ChangelogByDateTestCase(TempDirTestCase):
+
+    def _set_username(self, username):
+        config_writer = self.repo.config_writer()
+        config_writer.set_value('user', 'name', username)
+        config_writer.release()
+
+    def setup(self):
+        super(ChangelogByDateTestCase, self).setup()
+        self.changelog = TestableGitChangelogByDate()
+        self.changelog.state.document.settings.env.srcdir = self.root
+        self.repo = Repo.init(self.root)
+        self._set_username('Test User')
+
+
+class TestGroupByDate(ChangelogByDateTestCase):
```

Comment:
These tests look good, but I'd also like to see one for the rebase/cherry-pick case, checking that the ordering is as we expect it to be.